### PR TITLE
fix(engine): make `fn` types always arrows

### PR DESCRIPTION
**Pull request by W95Psp**
Now functions *always* have a arrow type of arity >=1.

### CODE REVIEW

File: examples/chacha20/proofs/fstar/extraction/Chacha20.fst [hunk 1]
```diff
@@ -21,15 +21,17 @@ let chacha20_line (a b d: usize) (s: u32) (m: t_Array u32 (sz 16))
...
   in
   let state:t_Array u32 (sz 16) =
-    Rust_primitives.Hax.update_at state d ((state.[ d ] <: u32) ^. (state.[ a ] <: u32) <: u32)
+    Rust_primitives.Hax.Monomorphized_update_at.update_at_usize state
+      d
+      ((state.[ d ] <: u32) ^. (state.[ a ] <: u32) <: u32)
   in
   let state:t_Array u32 (sz 16) =
-    Rust_primitives.Hax.update_at state
+    Rust_primitives.Hax.Monomorphized_update_at.update_at_usize state
```
Comment (karthikbhargavan) [inline examples/chacha20/proofs/fstar/extraction/Chacha20.fst:34]: Was this code out of date?

File: tests/attributes/src/lib.rs [hunk 2]
```diff
@@ -4,7 +4,7 @@ use hax_lib_macros as hax;
 const u32_max: u32 = 90000;
 
 #[hax::requires(x > 10 && y > 10 && z > 10 && x + y + z < u32_max)]
-#[hax::ensures(|result| result > 32)]
+#[hax::ensures(|result| hax_lib::implies(true, || result > 32))]
```
Comment (karthikbhargavan) [inline tests/attributes/src/lib.rs:7]: IS the F* code for this checked in?

**Review by karthikbhargavan — APPROVED** [review]

File: examples/chacha20/proofs/fstar/extraction/Chacha20.fst (hunk 1 above)
Comment (W95Psp) [reply examples/chacha20/proofs/fstar/extraction/Chacha20.fst:34]: Yes, the CI just discard the F* snapshot, so there is no policy regarding how up-to-date things are

File: tests/attributes/src/lib.rs (hunk 2 above)
Comment (W95Psp) [reply tests/attributes/src/lib.rs:7]: I added a snapshot this test crate: https://github.com/hacspec/hax/pull/388/commits/de3ed5fffb39ee91c7751c027afdd9b1e993f480

**Comment by W95Psp** [comment]
@karthikbhargavan: this branch makes Kyber go through, after promoting by hand some `<: bool` and `&&` to `<: Type0` and `/\`.
I also had to change `val simd256_support: bool` -> `val simd256_support: unit -> bool` in `Libcrux_platform.fsti`

**Pull request merged**